src/Neo/SmartContract/Native/Notary.cs

```diff
@@ -0,0 +1,342 @@
+        /// <param name="from">From Account</param>
+        /// <param name="to">To Account</param>
+        /// <returns>Whether withdrawal was successfull.</returns>
+        [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.States)]
+        private async ContractTask<bool> Withdraw(ApplicationEngine engine, UInt160 from, UInt160 to)
+        {
+            if (!engine.CheckWitnessInternal(from)) return false;
+            var receive = to is null ? from : to;
+            var deposit = GetDepositFor(engine.SnapshotCache, from);
+            if (deposit is null) return false;
+            if (Ledger.CurrentIndex(engine.SnapshotCache) < deposit.Till) return false;
+            RemoveDepositFor(engine.SnapshotCache, from);
+            if (!await engine.CallFromNativeContractAsync<bool>(Hash, GAS.Hash, "transfer", Hash.ToArray(), receive.ToArray(), deposit.Amount, StackItem.Null))
+            {
+                throw new InvalidOperationException(string.Format("Transfer to {0} has failed", receive.ToString()));
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// GetMaxNotValidBeforeDelta is Notary contract method and returns the maximum NotValidBefore delta.
+        /// </summary>
+        /// <param name="snapshot">DataCache</param>
+        /// <returns>NotValidBefore</returns>
+        [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates)]
+        public uint GetMaxNotValidBeforeDelta(IReadOnlyStore snapshot)
+        {
+            return (uint)(BigInteger)snapshot[CreateStorageKey(Prefix_MaxNotValidBeforeDelta)];
+        }
+
+        /// <summary>
+        /// SetMaxNotValidBeforeDelta is Notary contract method and sets the maximum NotValidBefore delta.
+        /// </summary>
+        /// <param name="engine">ApplicationEngine</param>
+        /// <param name="value">Value</param>
+        [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.States)]
+        private void SetMaxNotValidBeforeDelta(ApplicationEngine engine, uint value)
+        {
+            if (value > engine.ProtocolSettings.MaxValidUntilBlockIncrement / 2 || value < ProtocolSettings.Default.ValidatorsCount)
+                throw new FormatException(string.Format("MaxNotValidBeforeDelta cannot be more than {0} or less than {1}",
+                 engine.ProtocolSettings.MaxValidUntilBlockIncrement / 2, ProtocolSettings.Default.ValidatorsCount));
+            if (!CheckCommittee(engine)) throw new InvalidOperationException();
+            engine.SnapshotCache.GetAndChange(CreateStorageKey(Prefix_MaxNotValidBeforeDelta))!.Set(value);
+        }
+
+        /// <summary>
+        /// GetNotaryNodes returns public keys of notary nodes.
+        /// </summary>
+        /// <param name="snapshot">DataCache</param>
+        /// <returns>Public keys of notary nodes.</returns>
+        private static ECPoint[] GetNotaryNodes(DataCache snapshot)
```

Comment:
Redundant, I Prefer to direct call the RoleManagement. As I see it is just called twice.

Reply:
> Redundant,

No, it's not. It's a helper that uses proper index, and it's easy to forget about this index when using direct call.

> As I see it is just called twice.

It is called twice in the current implementation. Other parts of Notary subsystem will also use this helper, but these parts are not yet implemented in C#.